overlord/snapstate/snapstate.go

```diff
@@ -520,6 +520,11 @@ func Update(st *state.State, name, channel string, revision snap.Revision, userI
 		channel = snapst.Channel
 	}
 
+	if !flags.JailMode {
```

Comment:
is this tested? also seems obscure tbh, it will not be marked classic but it will be marked jailmode